src/emeus-constraint-layout.c

```diff
@@ -1407,7 +1407,12 @@ create_child_constraint (EmeusConstraintLayout      *layout,
   /* Alternatively, if it's not a constant value, we find the variable
    * associated with it
    */
-  if (constraint->source_object != NULL)
+  if (constraint->source_object == NULL ||
+      (EMEUS_IS_CONSTRAINT_LAYOUT (constraint->source_object) && constraint->source_object == layout))
+    {
+      attr2 = get_layout_attribute (layout, constraint->source_attribute);
+    }
+  else
```

Comment:
If we do this:

```
  else if (EMEUS_IS_CONSTRAINT_LAYOUT_CHILD (constraint->source_object))
    {
      ...
    }
```

then what happens to:

```
      else
        source_child = (EmeusConstraintLayoutChild *) gtk_widget_get_parent (constraint->source_object);
```

Are we replacing the target_object for the LayoutChild somewhere before it gets here?